### ISSUE TITLE
`donor_organism.organism_age` has a broken regex 

### ISSUE THREAD

**Issue by achave11-ucsc** [HumanCellAtlas/metadata-schema#1301]
https://github.com/HumanCellAtlas/metadata-schema/blob/3f9613c484ba72b003fa6100a5a10797f69a7adb/json_schema/type/biomaterial/donor_organism.json#L93

Actual behavior: The strings  `0-.`, `0.-`, `0..` are accepted by the regex.

Expected behavior: The regex should reject these strings.

**Comment by ESapenaVentura** [comment]
Hi @achave11 ! Thanks for noticing this.

I agree that these strings shouldn't be allowed. 

Would a regex like `^[0-9]+(\\.?[0-9]+)*(-?[0-9]+)*(\\.?[0-9]+)*$` be more fitting?

I have checked and it doesn't match with any of your examples.

Please let us know if you think this would not be a proper sustitution regex

**Comment by mshadbolt** [comment]
@ESapenaVentura would this work: `^[0-9]{1,3}?(-[0-9]{1,3})?$`

**Comment by ESapenaVentura** [comment]
Depends if we are only accepting integers as age or we are also accepting fractions (e.g. 14.5 years)

**Comment by mshadbolt** [comment]
right, i didn't realise we accepted fractions as it isn't specified in the description/guidelines or examples. I would propose we update these too. Though I do think we can limit the numbers to up to 3 as i don't think we would ever have an age in the thousands...?
I also changed the `*` to match either 0 or 1 matches, which I think is preferable to 0 or more.

I also put a positive look behind so that things like 34.4.4 wouldn't match. Let me know if you think it is ok. (do we know if positive look behind works in json schema validation?)

`^[0-9]{1,3}(\\.?[0-9]{1,2})?(-[0-9]{1,3})?((?<=-[0-9]{1,3})(\\.[0-9]{1,2}))?$`

Alternative could be to use the anyOf and have two separate patterns, one for single age and one for an age range. Perhaps I'll test if this works first if we are happy to try it.

**Comment by mshadbolt** [comment]
Further conversation with @ESapenaVentura we figured out it should be:
`^\\d{1,3}(\\.\\d+)?(-\\d{1,3}(\\.\\d+)?)?$`
